raft/raft.go

```diff
@@ -565,15 +565,25 @@ func (r *raft) Step(m pb.Message) error {
 	case m.Term > r.Term:
 		lead := m.From
 		if m.Type == pb.MsgVote {
+			if r.state == StateFollower && r.checkQuorum && r.electionElapsed < r.electionTimeout {
```

Comment:
this is also true for leader? right? also let's put the conditional var checking first

so
``` go
if r.checkQuorum && r.state != StateCandidate && r.electionElapsed < r.electionTimeout 
```